### PR TITLE
Added GNMT Example

### PR DESCRIPTION
**Pull request by levzlotnik**
Added the repository from `mlperf/training`

### CODE REVIEW

**Comment by levzlotnik** [comment]
@nzmora @guyjacob 
Please take a look at the current version of the notebook. The quantization is quite accurate on this one

File: distiller/quantization/range_linear.py [hunk 1]
```diff
@@ -875,12 +1005,25 @@ def replace_embedding(module, name, qbits_map, fp16=fp16):
...
-        self.replacement_factory[distiller.modules.EltwiseAdd] = partial(
+        factory_eltwiseadd = partial(
             replace_non_param_layer, RangeLinearQuantEltwiseAddWrapper)
-        self.replacement_factory[distiller.modules.EltwiseMult] = partial(
+        factory_eltwisemult = partial(
             replace_non_param_layer, RangeLinearQuantEltwiseMultWrapper)
+        factory_matmul = partial(
+            replace_non_param_layer, RangeLinearQuantMatmulWrapper)
+
+        update_wrapper(factory_concat, replace_non_param_layer)
```
[the remark this answers is not in the document]
Comment (guyjacob) [reply distiller/quantization/range_linear.py]: To be clear, I'm referring to the use of `update_wrapper`

[the remark this answers is not in the document]
Comment (levzlotnik) [reply distiller/quantization/range_linear.py]: `update_wrapper` allows to move some metadata from the original function to the wrapper function, like `__name__` or `__doc__`.  This in turn allows us to distinguish between the different functions in the `replacement_factory` when we view it from outside.  
No functional use, only clarification as done in one of the cells in `quantize_gnmt.ipynb`.

File: distiller/quantization/range_linear.py [hunk 2]
```diff
@@ -758,6 +855,39 @@ def forward(self, input):
         return out_f
 
 
+class RangeLinearQuantSoftmaxWrapper(RangeLinearQuantWrapper):
```
[the remark this answers is not in the document]
Comment (levzlotnik) [reply distiller/quantization/range_linear.py]: No good reason, will remove.

**Pull request merged**